### PR TITLE
feat: AnimatedSwitcher

### PR DESCRIPTION
**Pull request by darryltec**
## Motivations

Switching between 2 components is jarring and even unnoticeable. This helps add a bit of delightness and emphasis to the component you're switching.

It is best used on elements that the user would interact with such as (but not limited to) buttons and icons.

### Added

- Added `AnimatedSwitcher` component
- Added `AnimatedSwitcher.Icon` component which has its own style of animation

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Review by zevpermack — COMMENTED** [review]
LGTM, I tested the prerelease where Quotemeal will be using the component and it works for our needs. I left a couple comments from an Atlantis user experience perspective.

File: packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
+import { Button } from "@jobber/components/Button";
+import { Text } from "@jobber/components/Text";
+import { AnimatedSwitcher } from ".";
+
+<Meta title="Components/AnimatedSwitcher" component={AnimatedSwitcher} />
+
+# AnimatedSwitcher
+
+Smoothly switch between two components that aim to add delight when the user
```
Comment (taylorvnoj) [inline packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx:11]: Love this description

Now that we have a better idea of documentation sections moving forward should we include some of the relevant ones?  I think Content Guidelines and/or Design and usage guidelines to be a bit more clear about how this is best used between buttons/icons

**Review by taylorvnoj — COMMENTED** [review]
Just a few comments - looks really great!

File: packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          }
+        />
+      );
+    }}
+  </Story>
+</Canvas>
+
+## Props
+
+<ArgsTable of={AnimatedSwitcher} story="AnimatedSwitcher" />
```
[the remark this answers is not in the document]
Comment (darryltec) [reply packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx:45]: oh could be!

File: packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+<Canvas>
+  <Story name="AnimatedSwitcher">
+    {args => {
+      const [switched, setSwitched] = useState(args.switched || false);
+      return (
+        <AnimatedSwitcher
+          {...args}
+          switched={switched}
+          initialChild={
+            <Button label="Mark complete" onClick={() => setSwitched(true)} />
```
[the remark this answers is not in the document]
Comment (darryltec) [reply packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx:27]: that's a storybook thing so no control over it, sadly

File: packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx (hunk 1 above)
Comment (darryltec) [reply packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx:11]: not at this moment. It's as simple as it goes so I don't think we need one just yet

File: packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zevpermack) [reply packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx:45]: I guess the flip side of this is if it is required does it need a default value?

[the remark this answers is not in the document]
Comment (darryltec) [reply packages/components/src/AnimatedSwitcher/AnimatedSwitcher.stories.mdx:45]: nope. that was a force of habit. already removed!!

**Pull request merged**